defaultNatsService.go

```diff
@@ -0,0 +1,346 @@
+package natsio
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/nats-io/nats.go"
+	"github.com/orchestd/dependencybundler/interfaces/configuration"
+	"github.com/orchestd/dependencybundler/interfaces/credentials"
+	"github.com/orchestd/dependencybundler/interfaces/log"
+	. "github.com/orchestd/servicereply"
+	"github.com/orchestd/servicereply/status"
+	"go.uber.org/fx"
+	"reflect"
+	"time"
+)
+
+var connect = func(natsUrl string, options ...nats.Option) (NatsConnection, error) {
+	return nats.Connect(natsUrl, options...)
+}
+
+func NewDefaultNatsService(lc fx.Lifecycle, config configuration.Config, logger log.Logger, credentials credentials.CredentialsGetter) NatsService {
+	return getDefaultService(lc, config, logger, func(serviceName string) nats.Option {
+		natsUser := credentials.GetCredentials().NatsUser
+		if natsUser == "" {
+			panic("NotificationService could not get credentials by key NatsUser")
+		}
+		natsPw := credentials.GetCredentials().NatsPw
+		if natsPw == "" {
+			panic("NotificationService could not get credentials by key NatsPw")
+		}
+		authOpt := nats.UserInfo(natsUser, natsPw)
+		return authOpt
+	})
+}
+
+func NewDefaultNatsServiceWithJWTAuth(lc fx.Lifecycle, config configuration.Config, logger log.Logger) NatsService {
+	return getDefaultService(lc, config, logger, func(serviceName string) nats.Option {
+		authOpt := nats.UserCredentials(serviceName + ".creds")
```

Comment:
this means we need to pack it... in the docker ?

Reply:
why? create file on deploy? Need to ask Shani.